File: internal/domain/workflow_run.go
package domain

import (
	"time"
)

const CollectionNameWorkflowRun = "workflow_run"

type WorkflowRun struct {
	Meta
	WorkflowId string                `db:"workflowRef" json:"workflowId"`
	Status     WorkflowRunStatusType `db:"status"      json:"status"`
	Trigger    WorkflowTriggerType   `db:"trigger"     json:"trigger"`
	StartedAt  time.Time             `db:"startedAt"   json:"startedAt"`
	EndedAt    time.Time             `db:"endedAt"     json:"endedAt"`
	Graph      *WorkflowGraph        `db:"graph"       json:"graph"`
	Error      string                `db:"error"       json:"error"`
}

type WorkflowRunStatusType string

const (
	WorkflowRunStatusTypePending    WorkflowRunStatusType = "pending"
	WorkflowRunStatusTypeProcessing WorkflowRunStatusType = "processing"
	WorkflowRunStatusTypeSucceeded  WorkflowRunStatusType = "succeeded"
	WorkflowRunStatusTypeFailed     WorkflowRunStatusType = "failed"
	WorkflowRunStatusTypeCanceled   WorkflowRunStatusType = "canceled"
)

File: ui/src/domain/workflowRun.ts
import { type WorkflowGraph, type WorkflowModel } from "./workflow";

export interface WorkflowRunModel extends BaseModel {
  workflowRef: string;
  status: string;
  trigger: string;
  startedAt: ISO8601String;
  endedAt: ISO8601String;
  graph?: WorkflowGraph;
  error?: string;
  outputs?: Array<{
    type: string;
    name: string;
    value: string;
    valueType: string;
  }>;
  expand?: {
    workflowRef?: Pick<WorkflowModel, "id" | "name" | "description">;
  };
}

export const WORKFLOW_RUN_STATUSES = Object.freeze({
  PENDING: "pending",
  PROCESSING: "processing",
  SUCCEEDED: "succeeded",
  FAILED: "failed",
  CANCELED: "canceled",
} as const);

export type WorkflorRunStatusType = (typeof WORKFLOW_RUN_STATUSES)[keyof typeof WORKFLOW_RUN_STATUSES];
